src/metabase/analytics/stats.clj

```diff
@@ -362,37 +348,64 @@
 
 ;;; Execution Metrics
 
-(defn- summarize-executions
-  "Summarize `executions`, by incrementing approriate counts in a summary map."
-  ([]
-   (summarize-executions (t2/reducible-query {:select [:executor_id :running_time [[:case
-                                                                                    [:= :error nil ] false
-                                                                                    [:= :error "" ] false
-                                                                                    :else true] :has_error]]
-                                              :from   [:query_execution]})))
-  ([executions]
-   (reduce summarize-executions {:executions 0, :by_status {}, :num_per_user {}, :num_by_latency {}} executions))
-  ([summary execution]
-   (-> summary
-       (update :executions u/safe-inc)
-       ;; MYSQL is returning true as 1, false as 0! what!
-       (update-in [:by_status (if (#{1 true} (:has_error execution))
-                                "failed"
-                                "completed")] u/safe-inc)
-       (update-in [:num_per_user (:executor_id execution)] u/safe-inc)
-       (update-in [:num_by_latency (bin-large-number (/ (:running_time execution) 1000))] u/safe-inc))))
-
-(defn- summarize-executions-per-user
-  "Convert a map of `user-id->num-executions` to the histogram output format we expect."
-  [user-id->num-executions]
-  (frequencies (map bin-large-number (vals user-id->num-executions))))
+(defn- execution-metrics-sql []
+  (let [thirty-days-ago (str "CAST('" (t/minus (t/offset-date-time) (t/days 30)) "' AS TIMESTAMP)")]
```

Comment:
Are you sure this shouldn't be `AS TIMESTAMP WITH TIME ZONE`?

Reply:
Fine catch. `TIMESTAMP WITH TIME ZONE` doesn't existing in MySQL so I was taking advantage of just using `TIMESTAMP`, which is common to all Postgres, H2, and MySQL. The database would assume the timestamp was in the DB's time zone. Now I see that doesn't account for the fact that the app DB might be in a different time zone to the JVM, in which case this would be incorrect. I can't see any docs that prohibit them having different time zones technically I think we should handle this case. I've updated the code to use database-specific SQL strings now for this timestamp.